railib/api.py

```diff
@@ -657,43 +649,83 @@ def _model(name: str, model: str) -> dict:
     }
 
 
-# Returns full list of models.
-def _list_models(ctx: Context, database: str, engine: str) -> dict:
-    tx = Transaction(database, engine, mode=Mode.OPEN)
-    rsp = tx.run(ctx, _list_action())
-    actions = rsp["actions"]
-    assert len(actions) == 1
-    action = actions[0]
-    models = action["result"]["sources"]
-    return models
-
-
 def create_database(ctx: Context, database: str, source: str = None) -> dict:
     data = {"name": database, "source_name": source}
     url = _mkurl(ctx, PATH_DATABASE)
     rsp = rest.put(ctx, url, data)
     return json.loads(rsp.read())
 
 
-def delete_model(ctx: Context, database: str, engine: str, model: str) -> dict:
-    tx = Transaction(database, engine, mode=Mode.OPEN, readonly=False)
-    actions = [_delete_model_action(model)]
-    return tx.run(ctx, *actions)
+# Returns full list of models.
+def list_models(ctx: Context, database: str, engine: str) -> List:
+    models = []
+    out_name = f'model{random.randint(0, sys.maxsize)}'
+    resp = exec(ctx, database, engine, f'def output:{out_name}[name] = rel:catalog:model(name, _)')
```

Comment:
why does this qualify the output relation?

Reply:
we qualify the output relation here to avoid collision with user predefined outputs

Reply:
👍 The JS sdk does something similar, that makes sense to me.